test/testhelper/client/k8s/client.go

```diff
@@ -0,0 +1,276 @@
+	}
+	return nodes, nil
+}
+
+func (c *Client) ListNodeIPByLabel(label string) ([]net.IP, error) {
+	var ips []net.IP
+	nodes, err := c.ListNodesByLabel(label)
+	if err != nil {
+		return nil, err
+	}
+	for _, node := range nodes.Items {
+		for _, v := range node.Status.Addresses {
+			if v.Type == v1.NodeInternalIP {
+				ips = append(ips, net.ParseIP(v.Address))
+			}
+		}
+	}
+	return ips, nil
+}
+
+func (c *Client) ListAllNamespacesPods() ([]*NamespacePod, error) {
+	namespaceList, err := c.listNamespaces()
+	if err != nil {
+		return nil, err
+	}
+	var namespacePodList []*NamespacePod
+	for _, ns := range namespaceList.Items {
+		pods, err := c.client.CoreV1().Pods(ns.Name).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get all namespace pods")
+		}
+		namespacePod := NamespacePod{
+			Namespace: ns,
+			PodList:   pods,
+		}
+		namespacePodList = append(namespacePodList, &namespacePod)
+	}
+
+	return namespacePodList, nil
+}
+
+func (c *Client) ListKubeSystemPodsStatus() (bool, error) {
+	pods, err := c.client.CoreV1().Pods("kube-system").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to get kube-system namespace pods")
+	}
+	// pods.Items maybe nil
+	if len(pods.Items) == 0 {
+		return false, nil
+	}
+	for _, pod := range pods.Items {
+		// pod.Status.ContainerStatus == nil because of pod contain initcontainer
+		if len(pod.Status.ContainerStatuses) == 0 {
+			continue
+		}
+		if !pod.Status.ContainerStatuses[0].Ready {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (c *Client) GetPodLog(nameSpace, podName string) (string, error) {
+	req := c.client.CoreV1().Pods(nameSpace).GetLogs(podName, &v1.PodLogOptions{})
+	podLogs, err := req.Stream(context.TODO())
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	str := buf.String()
+	return str, nil
+}
+
+func (c *Client) GetPodEvents(nameSpace, podName string) ([]v1.Event, error) {
+	events, err := c.client.CoreV1().
+		Events(nameSpace).
+		List(context.TODO(), metav1.ListOptions{FieldSelector: "involvedObject.name=" + podName, TypeMeta: metav1.TypeMeta{Kind: "Pod"}})
+	if err != nil {
+		return nil, err
+	}
+	return events.Items, nil
+}
+
+func (c *Client) getPodReadyStatus(pod v1.Pod) error {
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == ReadyStatus {
+			if condition.Status == TRUE {
+				return nil
+			}
+		}
+	}
+	return &checker.NotFindReadyTypeError{}
+}
+
+func (c *Client) GetNotReadyPodEvent() (map[string][]EventPod, error) {
+	namespacePodList, err := c.ListAllNamespacesPods()
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string][]EventPod)
+	for _, podNamespace := range namespacePodList {
+		for _, pod := range podNamespace.PodList.Items {
+			if err := c.getPodReadyStatus(pod); err != nil {
```

Comment:
if err := c.getPodReadyStatus(pod); err == nil {
  continue
}

In programming, we need to pay attention to two things:
1. return fast
2. avoid too many depth